sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedPartitionReader.scala

```diff
@@ -244,7 +248,7 @@ private case class ChangeFeedPartitionReader
         assert(node.lsn != "", "Change feed responses must have non empty _lsn.")
         val nextLsn = SparkBridgeImplementationInternal.toLsn(node.lsn)
 
-        nextLsn <= endLsn
+        nextLsn <= endLsn || this.iterator.hasMoreChangesToProcess.get()
```

Comment:
I think the validateNextLsn should be part of the this.iterator.hasNext checking logic. If we have reached the endLSN for the current page, then move on to get the next page

hmm, also not quite understand why need to introduce a new flag hasMoreChangesToProcess

Reply:
Moved the logic to the hasNext of the iterator and removed the flag.